### PR TITLE
[cleanup] Consolidate profile-loading error messages

### LINKED ISSUE #676: [cleanup] Consolidate profile-loading error messages
## Why / Context
Profile-related failures currently emit inconsistent message shapes across loaders and validators, increasing triage friction.

## Scope
**In scope**
- Introduce a shared formatter/helper for profile-loading error payloads.
- Apply it across app loader and preflight profile checks.
- Preserve error categories while standardizing wording/fields.

**Out of scope**
- Changing validation rules or profile business logic.
- Adding new profile sources.

Constraints:
- Keep changes small/mergeable.
- Deterministic tests only.
- Avoid touching unrelated generated artifacts unless required.

## Acceptance Criteria (required)
- [ ] Profile-loading failures use one normalized message shape across call sites.
- [ ] Error payload includes actionable reason + remediation hint.
- [ ] Existing callers parsing category/severity do not break.
- [ ] Unit tests cover representative failure classes and message formatting.

## Implementation Notes / Pointers
**Likely files / modules:**
- `src/gpt_trader/app/config/profile_loader.py`
- `src/gpt_trader/preflight/checks/profile.py`
- `src/gpt_trader/validation/config_validators.py`

**Related tests:**
- `tests/unit/gpt_trader/app/config/test_profile_loader.py`
- `tests/unit/gpt_trader/preflight/test_checks_profile.py`
- `tests/unit/gpt_trader/validation/test_config_validators.py`

**Edge cases to handle:**
- Missing file vs malformed payload errors should remain distinguishable.
- Multiline upstream exceptions should be safely normalized.
- Normalization should not leak secrets from environment-derived config.

## Commands (local)
- `make fmt`
- `make lint`
- `uv run pytest -q tests/unit/gpt_trader/app/config/test_profile_loader.py tests/unit/gpt_trader/validation/test_config_validators.py`

## PR Requirements
- PR title should match the issue.
- PR body must include: `Fixes #<issue-number>`
- CI must be green; if Agent Artifacts Freshness fails, run `uv run agent-regenerate`, then commit `var/agents/...` updates.

## Codex-Ready Checklist (for the issue creator)
- [x] Clear acceptance criteria
- [x] At least one file pointer
- [x] Commands included
- [x] No ambiguous "do the right thing" language

### PR DESCRIPTION
**Pull request by Solders-Girdles**
Fixes #676

(Automated PR created by OpenClaw PR finisher)
Branch: codex/676-cleanup-consolidate-profile-loading-erro
Commit: b55a0cacbb643f7714dca62be05920768cd5f288

### CODE REVIEW

File: src/gpt_trader/preflight/checks/profile.py [hunk 1]
```diff
@@ -46,12 +50,23 @@ def check_profile_configuration(checker: PreflightCheck) -> bool:
...
             return True
-
         except Exception as exc:
-            checker.log_error(f"Failed to parse profile: {exc}")
+            payload = profile_yaml_parse_error_payload(
+                profile=checker.profile,
+                path=profile_path,
+                exception=exc,
+            )
+            checker.log_error("Failed to parse profile", details=payload)
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/preflight/checks/profile.py]: Addressed in 158ce048: parse failures now keep the exception text in the user-facing preflight error (`Failed to parse profile: {exc}`), and I added a unit assertion to lock that behavior (`test_fails_when_yaml_parse_fails`).
